### PR TITLE
[WIP] Use lerna for packages

### PR DESCRIPTION
**Pull request by harshjv**
- [x] Create npm package for each provider & modules within CAL
- [x] Single build process for all the packages
- [x] Add/port unit tests
- [x] Add/port integration tests
- [x] Add readme to all the packages
- [x] Update examples
- [x] Add source maps
- [ ] Assign proper versioning
- [ ] Add documentation for all the packages

## Instructions

```
npm install
npm run bootstrap # do npm install in each package & link dependencies
npm run build # build all the packages
```

### CODE REVIEW

File: packages/ethereum-erc20-provider/lib/EthereumERC20Provider.js [hunk 1]
```diff
@@ -1,20 +1,25 @@
...
     this._contractAddress = ensureAddressStandardFormat(contractAddress)
   }
 
   generateErc20Transfer (to, value) {
-    // function transfer(address to, uint tokens) public returns (bool success)
-    const functionSignature = 'a9059cbb'
     const encodedAddress = padHexStart(ensureAddressStandardFormat(to), 64)
     const encodedValue = padHexStart(value.toString(16), 64)
-    return functionSignature + encodedAddress + encodedValue
+    return SOL_TRANSFER_FUNCTION.substring(2) + encodedAddress + encodedValue
```
Comment (monokh) [inline packages/ethereum-erc20-provider/lib/EthereumERC20Provider.js:22]: Strange to see these changes here. Just checking if this is some git problem?

File: packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.js [hunk 2]
```diff
@@ -57,11 +61,11 @@ export default class EthereumLedgerProvider extends LedgerProvider {
     const path = address.derivationPath
 
     let txData = {
-      to: to ? ensureHexEthFormat(to) : undefined,
+      to: to ? ensureHexEthFormat(to) : null,
       from: ensureHexEthFormat(from),
-      value: BigNumber(value).toNumber(),
+      value: ensureHexEthFormat(BigNumber(value).toString(16)),
```
Comment (monokh) [inline packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.js:66]: Similarly with these changes. Thought that they were already in dev.

**Review by grandsmarquis — APPROVED** [review]
LGTM! :+1:

File: packages/ethereum-erc20-provider/lib/EthereumERC20Provider.js (hunk 1 above)
Comment (harshjv) [reply packages/ethereum-erc20-provider/lib/EthereumERC20Provider.js:22]: No. I have made these changes. Making functional signature more readable.

File: packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.js (hunk 2 above)
Comment (harshjv) [reply packages/ethereum-ledger-provider/lib/EthereumLedgerProvider.js:66]: i couldn't cherry pick jules' commits so i had to copy a few changes. only four lines were there.

**Pull request merged**